service/resource/endpoint/create.go

```diff
@@ -1,11 +1,73 @@
 package endpoint
 
-import "context"
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
 
 func (r *Resource) GetCreateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	return nil, nil
+	currentEndpoints, err := toEndpointSlice(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	desiredEndpoints, err := toEndpointSlice(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	createState := []Endpoint{}
+	for _, currentEndpoint := range currentEndpoints {
+		if !containsEndpoint(createState, currentEndpoint) {
+			createState = append(createState, currentEndpoint)
+		}
+	}
+	for _, desiredEndpoint := range desiredEndpoints {
+		if !containsEndpoint(createState, desiredEndpoint) {
+			createState = append(createState, desiredEndpoint)
+		}
+	}
+
+	return createState, nil
 }
 
 func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState interface{}) error {
+	endpointsToCreate, err := toEndpointSlice(createState)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	var serviceName string
+	var serviceNamespace string
+	if len(endpointsToCreate) > 0 {
+		serviceName = endpointsToCreate[0].ServiceName
```

Comment:
We can do this as all `ServiceName`s here should be the same by now.

Reply:
IMO this means `createState` has wrong type. Probably should be something like

```
struct {
  	IPs              []string
	ServiceName      string
	ServiceNamespace string
}

Reply:
Yeah I was starting to think the same, going to fix this.